libs/ngxtension/click-outside/src/click-outside.ts

```diff
@@ -0,0 +1,62 @@
+import {
+	Directive,
+	ElementRef,
+	EventEmitter,
+	inject,
+	Injectable,
+	NgZone,
+	Output,
+} from '@angular/core';
+
+import type { OnInit } from '@angular/core';
+import { injectDestroy } from 'ngxtension/inject-destroy';
+import { fromEvent, Subject, takeUntil } from 'rxjs';
+
+/*
+ * This service is used to detect clicks in the document.
+ * It is used by the clickOutside directive.
+ */
+@Injectable({ providedIn: 'root' })
+export class DocumentClickService {
```

Comment:
suggestion: this can easily be a CIF instead of a full-blown class, especially we don't even expose this as a public API for `click-outside`

```ts
const [injectDocumentClick] = createInjectionToken(() => {
    const click$ = new Subject<MouseEvent>();
    const [ngZone, document] = [inject(NgZone), inject(DOCUMENT)];
    ngZone.runOutsideAngular(() => {
        fromEvent(document, 'click').subscribe(click$);
    });
    return click$;
})

// then in the directive
export class ClickOutside {
    documentClick$ = injectDocumentClick();
}
```
})

Reply:
Nice suggestion!